app/helpers/application_helper.rb

```diff
@@ -54,6 +54,15 @@ def render_index_grid(grid)
     render "admin/common/grid", no_records: no_records, grid: grid
   end
 
+  def render_loan_grid(grid)
+    if grid.all_pages_records.count == 1 && grid.filtered_by == ['id']
```

Comment:
Maybe it would make sense to generalize this? I am going to guess the other grids would have the same attributes, but there are none with ID filters to test on.